src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/ParameterCapturingService.cs

```diff
@@ -1,81 +1,270 @@
+                }
             }
-            catch
+        }
+
+        private void OnStopMessage(EmptyPayload _)
+        {
+            if (_stopRequests?.Writer.TryWrite(true) != true)
             {
-                // TODO: Log
+                if (!IsAvailable())
+                {
+                    _eventSource.ServiceNotAvailable(_notAvailableReason!.Value, _notAvailableDetails);
+                    return;
+                }
+                else
+                {
+                    // The channel is full which is OK as stop requests aren't tied to a specific operation.
+                }
             }
         }
 
-        public void StopCapturing()
+        private void StartCapturing(StartCapturingParametersPayload request)
         {
-            if (!_isAvailable)
+            if (!IsAvailable())
             {
                 throw new InvalidOperationException();
             }
 
-            _probeManager?.StopCapturing();
+            MethodResolver resolver = new();
+            List<MethodInfo> methods = new(request.Methods.Length);
+            List<MethodDescription> methodsFailedToResolve = new();
+
+            for (int i = 0; i < request.Methods.Length; i++)
+            {
+                MethodDescription methodDescription = request.Methods[i];
+
+                List<MethodInfo> resolvedMethods = resolver.ResolveMethodDescription(methodDescription);
+                if (resolvedMethods.Count == 0)
+                {
+                    methodsFailedToResolve.Add(methodDescription);
+                }
+
+                methods.AddRange(resolvedMethods);
+            }
+
+            if (methodsFailedToResolve.Count > 0)
+            {
+                UnresolvedMethodsExceptions ex = new(methodsFailedToResolve);
+                _logger!.LogWarning(ex.Message);
+                throw ex;
+            }
+
+            _probeManager!.StartCapturing(methods);
+            _logger!.LogInformation(
+                ParameterCapturingStrings.StartParameterCapturingFormatString,
+                request.Duration,
+                request.Methods.Length);
         }
 
-        public void StartCapturing(IList<MethodInfo> methods)
+        private void StopCapturing()
         {
-            if (!_isAvailable)
+            if (!IsAvailable())
             {
-                throw new InvalidOperationException();
+                return;
+            }
+
+            _logger!.LogInformation(ParameterCapturingStrings.StopParameterCapturing);
+            _probeManager!.StopCapturing();
+        }
+
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            if (!IsAvailable())
+            {
+                return;
+            }
+
+            void tryToStopCapturing()
+            {
+                try
+                {
+                    StopCapturing();
+                    _eventSource.CapturingStop();
```

Comment:
Why not make this part of the `StopCapturing` method so that it is always evented no matter what tries to stop the capture.